aptos-move/framework/aptos-token/sources/listing.move

```diff
@@ -0,0 +1,243 @@
+module aptos_token::listing {
+    use std::signer;
+    use std::string::String;
+    use aptos_std::table::{Self, Table};
+    use aptos_token::token::{Self, TokenId};
+    use aptos_std::guid::{Self, ID};
+    use aptos_framework::account;
+    use aptos_framework::event::{Self, EventHandle};
+
+
+    const EOWNER_NOT_HAVING_ENOUGH_TOKEN: u64 = 1;
+    const ELISTING_NOT_EXIST:u64 = 2;
+
+    /// immutable struct for recording listing info.
+    struct Listing<phantom CoinType> has drop, store {
+        id: ID,
+        owner: address,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        instant_sale: bool, // true for marketplace and false for auction
```

Comment:
could this be both?

Reply:
No, if instant_sale is true, the sale immediately happens if someone bid higher than the listed price. We don't need to wait till the end of the auction.